.github/workflows/deploy.yml

```diff
@@ -0,0 +1,61 @@
+name: Docker
+
+on:
+  push:
+    branches: [ main ]
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+
+jobs:
+  build-and-push-image:
+    name: Build and publish
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+  deploy:
+    needs: [build-and-push-image]
+    name: Deploy to server
+    runs-on: ubuntu-latest
+    steps:
+    - name: Deploy to server
+      uses: fifsky/ssh-action@master
```

Comment:
We should use the more popular `https://github.com/appleboy/ssh-action` for the sake of security. This action is also proven work well in our production environment.
The deploy action will require other minor modifications to suite this change.

Reply:
I actually tried to use that but I ran into problems with authentication while testing with my own server (and the `fifsky/ssh-action` worked). If the action though works in the production I do not see any problems with that.

Reply:
Yup! This GH-action is currently in use for at least [testausserveri/testausapis](https://github.com/Testausserveri/testausapis/) and has proven to work well. You may reffer to it's [deploy.yml ](https://github.com/Testausserveri/testausapis/blob/main/.github/workflows/deploy.yml)